app/main.py

```diff
@@ -1,8 +1,34 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        Person.people[name] = self
+
+
+def create_person_list(people: list) -> list[Person]:
+    result = []
+    for person_data in people:
+        name = person_data["name"]
+        age = person_data["age"]
+
+        new_person = Person(name, age)
+        result.append(new_person)
+
+        wife_name = person_data.get("wife")
+        if wife_name:
+            wife = Person.people.get(wife_name)
+            if wife:
+                new_person.wife = wife
+                wife.husband = new_person
+
+        husband_name = person_data.get("husband")
+        if husband_name:
+            husband = Person.people.get(husband_name)
```

Comment:
Similar to the previous comment, retrieving a `Person` instance using the husband's name assumes name uniqueness. This could lead to incorrect associations if names are not unique.